pxtblocks/fields/field_ledmatrix.ts

```diff
@@ -80,17 +81,111 @@ export class FieldMatrix extends Blockly.Field implements FieldCustom {
+                }
+                break;
+            }
+            case "KeyD":
+            case "ArrowRight": {
+                if (x !== this.cells.length - 1) {
+                    this.selected = [x + 1, y]
+                } else if (y !== this.matrixHeight - 1) {
+                    this.selected = [0, y + 1]
+                }
+                break;
+            }
+            case "Home": {
+                if (ctrlCmd) {
+                    this.selected = [0, 0]
+                } else {
+                    this.selected = [0, y]
+                }
+                break;
+            }
+            case "End": {
+                if (ctrlCmd) {
+                    this.selected = [this.matrixWidth - 1, this.matrixHeight - 1]
+                } else {
+                    this.selected = [this.matrixWidth - 1, y]
+                }
+                break;
+            }
+            case "Enter":
+            case "Space": {
+                this.toggleRect(x, y, !this.cellState[x][y]);
+                break;
+            }
+            case "Escape": {
+                (Blockly.getMainWorkspace() as Blockly.WorkspaceSvg).markFocused();
+                return;
+            }
+            default: {
+                return
+            }
+        }
+        this.setFocusIndicator(this.cells[this.selected[0]][this.selected[1]]);
+        this.elt.setAttribute('aria-activedescendant', `${this.sourceBlock_.id}:${this.selected[0]}${this.selected[1]}`);
+        e.preventDefault();
+        e.stopPropagation();
+    }
+
+    private blurHandler() {
+        this.elt.removeEventListener("keydown", this.keyHandler)
+        this.elt.removeEventListener("blur", this.blurHandler)
+        if (this.selected) {
+            this.setFocusIndicator();
+            this.selected = undefined;
+        }
+        this.elt.removeAttribute('aria-activedescendant');
+    }
+
+    private setFocusIndicator(cell?: SVGRectElement) {
+        this.cells.forEach(cell => cell.forEach(cell => cell.nextElementSibling.firstElementChild.classList.remove("selected")));
+        if (cell) {
+            cell.nextElementSibling.firstElementChild.classList.add("selected");
+        }
+    }
+
     /**
      * Show the inline free-text editor on top of the text.
      * @private
      */
     showEditor_() {
-        // Intentionally left empty
+        this.selected = [0, 0];
+        this.setFocusIndicator(this.cells[0][0])
+        this.elt.setAttribute('aria-activedescendant', this.sourceBlock_.id + ":00");
+        this.elt.focus();
     }
 
     private initMatrix() {
         if (!this.sourceBlock_.isInsertionMarker()) {
-            this.elt = pxsim.svg.parseString(`<svg xmlns="http://www.w3.org/2000/svg" id="field-matrix" />`);
+            this.elt = pxsim.svg.parseString(`<svg xmlns="http://www.w3.org/2000/svg" id="field-matrix" class="blocklyMatrix" tabindex="-1" role="grid" aria-label="${lf("LED matrix")}" />`);
```

Comment:
Let's go with "grid" as the user facing text.

Reply:
[64b3989](https://github.com/microbit-matt-hillsdon/pxt/pull/11/commits/64b3989c4040faf97bc73943d4161a6bd3064eba)